index.html

```diff
@@ -495,6 +495,139 @@ <h2>
         </p>
       </section>
     </section>
+    <section data-cite="image-resource">
+      <h2>
+        Manifest override objects
+      </h2>
+      <p>
+        Each <dfn>manifest override object</dfn> is a generic object value
+        that allows for certain manifest properties to be overridden within
+        a particular context.
+      </p>
+      <p>The structure of a [=manifest override object=] is as follows:</p>
+      <pre class="json">
+        {
+          "context_key": {
+            "property": "new value"
+          }
+        }
+      </pre>
+      <p>
+        Each manifest property that accepts a [=manifest override object=]
+        as its value will define the contexts it supports and which
+        properties it supports overriding. User agents MUST ignore any contexts
+        that are not supported by the property as well as any override
+        properties not explicitly allowed within it.
+      </p>
+      <p>
+        When the manifest property’s context is applicable, the value of each
+        allowable override will be used in place of the original value defined
+        in the Manifest.
+      </p>
+      <p>
+        Redefined array items will be overridden in the order they are authored.
+        When redefining objects (e.g., [=manifest/shortcuts=], [=manifest/icons=]),
+        authors will only be able to redefine specific properties of that object.
+        In order to ensure all overrides are applied correctly, the order must
+        match the original array (i.e., each [=manifest/shortcut=] must be redefined
+        in order, as must their icons, if they also require re-definition).
+      </p>
+        Redefined array items must also be equal in number to the array being
+        overridden. If there is a mismatch in the number of items in either array,
+        any excess items will be ignored. This is only an issue if the original
+        array has more items than the override array, because any excess items
+        within the original array will not be re-defined.
+      </p>
+      <p>
+        When there is a conflict because two different properties are attempting
+        to override the same value in their respective active contexts, the one
+        defined last will win. By way of example, consider the following:
+      </p>
+      <pre class="json">
+        {
+          "lang": "en-US",
+          "icons": [
+            {
+              "src": "icon.png",
+              "sizes": "128x128",
+              "type": "image/png"
+            }
+          ],
+          "translations": {
+            "es": {
+              "icons": [
+                {
+                  "src": "icon-es.png"
+                }
+              ]
+            }
+          },
+          "user_preferences": {
+            "color_scheme_dark": {
```

Comment:
This is nested one level deeper now, see https://github.com/WICG/manifest-incubations/blob/gh-pages/user-preferences-explainer.md for the latest version.

Reply:
I did see that and wanted to discuss: Would the manifest override construct hold up if there’s another level of nesting? I think the property > context = overrides makes sense, but in the revised user prefs model, it’s more like property > property > value = overrides. Feels inconsistent. In my view of these two features, having a single context to map to makes more sense.

Couldn’t the same thing be achieved by making the context "color_scheme_VALUE" (as we had it and how I used it) or "color_scheme: value" (or something similar)? What does the additional nesting give us in exchange for the cognitive overhead of having to manage more curly braces?

Reply:
IIRC this kind of nesting was strongly recommended by the TAG.

Reply:
Yes, the extra nesting was wanted by TAG.

I wonder if we need to define this as property > context > overrides? Could we just define the overrides part? Maybe we will also want to use this in future fields that don't fit this nesting?

Reply:
I would love a pointer to that TAG conversation (just for backstory). I thought on it a bit more and I think I have reasoned it out. `user_preferences` has its own set of properties and those properties take a manifest override object. So the structure ends up being more like property > sub-property > context = overrides. It still feels unnecessarily complex (on another thread @marcoscaceres was actually pushing for flatter Manifest structures), but I can make it work from a spec standpoint.

Reply:
This is TAG review here: https://github.com/w3ctag/design-reviews/issues/696

I don't have a strong preference as to which structure we use, as long as the TAG reviewers are on board.

Reply:
Thanks for that. I'm good with the adjusted structure and I think we just define them (e.g., `color_scheme`) as sub-properties whose values are a manifest override object. That does give us the extensibility for contrast, etc. too. Thanks for giving me the space to think through it a little more.

Can you update my conflict example accordingly or do you want me to do it?

Reply:
Thanks for thinking that through, that sounds good to me.

Yep, I can update that.

Reply:
Awesome thanks!

Reply:
As an ex-TAG member, I'm not sure the TAG's opinions has more weight than anyone else's. Their guidance is always valuable and welcome, but we are free to explore a range of solutions that are best for developers.